contracts/reserve/ExchangeHelper.sol

```diff
@@ -301,35 +301,37 @@ contract ExchangeHelper is DAOUpgradeableContract {
 			nameService.getAddress("UNISWAP_ROUTER")
 		);
 		address[] memory path = new address[](2);
-
+		address wETH = uniswapContract.WETH();
 		uint256[] memory swap;
-		if (address(_token) == address(0x0)) {
-			if (_isBuy) {
-				path[0] = uniswapContract.WETH();
-				path[1] = daiAddress;
-				swap = uniswapContract.swapExactETHForTokens{
-					value: _tokenAmount
-				}(_minDAIAmount, path, address(this), block.timestamp);
-				return swap;
-			} else {
-				path[0] = daiAddress;
-				path[1] = uniswapContract.WETH();
-				swap = uniswapContract.swapExactTokensForETH(
-					_tokenAmount,
-					_minTokenReturn,
-					path,
-					_receiver,
-					block.timestamp
-				);
-				return swap;
-			}
+		path[0] = _inputToken == address(0x0) ? wETH : _inputToken;
+		path[1] = _outputToken == address(0x0) ? wETH : _outputToken;
+
+		if (path[0] == wETH) {
```

Comment:
if(inputToken== (0x0))
because if user actually uses real WETH it will break

Reply:
no then _inputToken would be wETH and according if it's buy or sell in the below if statements it would go to proper function of uniswap

Reply:
Changed it @sirpy